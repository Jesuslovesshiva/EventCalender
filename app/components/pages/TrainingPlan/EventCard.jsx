import React from 'react';
// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
// import { faMapMarkerAlt, faUser, faClock, faFile } from '@fortawesome/free-solid-svg-icons';

const EventCard = ({ time, title, attendees, activities, attachments }) => {
  const renderAttachments = () => {
    return attachments.map((attachment, index) => (
      <div key={index} className="px-1">
        {/* <FontAwesomeIcon icon={faFile} className="text-black" /> */}
      </div>
    ));
  };

  const renderActivities = () => {
    return activities.map((activity, index) => (
      <div key={index} className="mb-4">
        <div className="text-black font-bold">
          {index + 1}. {activity.title}
        </div>
        <div className="bg-white rounded-full px-2 py-1 mt-2 flex items-center space-x-1 border border-black">
          {/* <FontAwesomeIcon icon={faClock} className="text-black" /> */}
          <span>{activity.duration} min</span>
          <div className="border-l border-black mx-1"></div>
          {/* <FontAwesomeIcon icon={faUser} className="text-black" /> */}
          <span>{activity.maxParticipants}</span>
        </div>
        <div className="text-black text-sm mt-2">{activity.description}</div>
      </div>
    ));
  };

  return (
    <div className="bg-gray-200 shadow-md rounded-lg p-4 w-full">
      <div className="flex items-center justify-between mb-2">
        <div className="text-black font-bold">{time}</div>
        <h3 className="text-lg font-bold text-black">{title}</h3>
        <div className="text-black">
          {/* <FontAwesomeIcon icon={faMapMarkerAlt} /> */}
        </div>
      </div>
      <hr className="border-black border-opacity-25 mb-2" />
      <div className="flex items-center justify-between mb-2">
        <div className="text-black">{attendees} participants</div>
        <div className="flex items-center space-x-1">{renderAttachments()}</div>
      </div>
      <hr className="border-black border-opacity-25 mb-2" />
      <div className="space-y-4">{renderActivities()}</div>
    </div>
  );
};

export default EventCard;
