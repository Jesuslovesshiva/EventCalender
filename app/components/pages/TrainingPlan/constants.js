const TRAINING_PLAN_MODULE_ID = 'TRAINING_PLAN'
const CALENDAR_MODULE_ID = 'TRAINING_PLAN_CALENDAR'

export {CALENDAR_MODULE_ID, TRAINING_PLAN_MODULE_ID}
