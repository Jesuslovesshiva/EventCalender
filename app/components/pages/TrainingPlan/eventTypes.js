const eventTypes = {
  training: {
    color: 'red',
    backgroundColor: 'bg-red-200'
  },
  team: {
    color: 'green',
    backgroundColor: '#D1FAE5'
  },
  game: {
    color: 'blue',
    backgroundColor: '#DBEAFE'
  }
}

export default eventTypes
