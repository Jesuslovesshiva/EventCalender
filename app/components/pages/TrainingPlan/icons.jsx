import {
  faLocationDot,
  faUser,
  faImage,
  faCirclePlay,
  faFileLines,
  faVideo,
  faClock,
  faPersonArrowUpFromLine,
  faLocationCrosshairs
} from '@fortawesome/free-solid-svg-icons'

export const icons = {
  locationDot: faLocationDot,
  user: faUser,
  image: faImage,
  circlePlay: faCirclePlay,
  fileLines: faFileLines,
  video: faVideo,
  clock: faClock,
  personArrowUpFromLine: faPersonArrowUpFromLine,
  target: faLocationCrosshairs
}

export default icons
