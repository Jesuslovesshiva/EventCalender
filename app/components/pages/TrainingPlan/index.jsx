import React from 'react'

import ModalRouter from '@/components/molecules/ModalRouter'
import PageHeader from '@/components/molecules/PageHeader'
import {
  TRAINING_PLAN_MODULE_ID
} from '@/components/pages/TrainingPlan/constants'
import {ROUTE_IDS} from '@/components/routes'
import {
  TRAINING_PLAN_BRANCH_ID,
  TRAINING_PLAN_ROUTE_IDS
} from '@/components/routes/training-plan'
import {getPathById} from '@/components/routes/utils'
import useEventResults from '@/hooks/event/use-event-results'
import EventResultsInjector from '@/injectors/event/ResultsInjector'

import messages from './messages'
import createRoutes from './modal-routes'

const BASE_PATH = getPathById([
  ROUTE_IDS.HOME,
  TRAINING_PLAN_BRANCH_ID,
  TRAINING_PLAN_ROUTE_IDS.ROOT
])
const ROUTES = createRoutes(BASE_PATH)

const TrainingPlan = () => {
  // eslint-disable-next-line no-unused-vars
  const [state, events] = useEventResults({id: TRAINING_PLAN_MODULE_ID, shouldLoadOnMount: true})

  return (
    <ModalRouter base={BASE_PATH} routes={ROUTES}>
      <EventResultsInjector />
      <div className="h-full flex flex-col">
        <PageHeader
          className="relative z-20"
          title={messages.pageTitle}
        />

        <div className="grow overflow-auto">
          <h2 className="font-heading-large">
            Have fun :)!
          </h2>
        </div>
      </div>
    </ModalRouter>
  )
}

export default React.memo(TrainingPlan)
