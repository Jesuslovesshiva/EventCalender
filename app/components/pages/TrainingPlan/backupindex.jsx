import React, {useState, useCallback} from 'react'
import ModalRouter from '@/components/molecules/ModalRouter'
import PageHeader from '@/components/molecules/PageHeader'
import {
  // CALENDAR_MODULE_ID,
  TRAINING_PLAN_MODULE_ID
} from '@/components/pages/TrainingPlan/constants'
import {ROUTE_IDS} from '@/components/routes'
import {
  TRAINING_PLAN_BRANCH_ID,
  TRAINING_PLAN_ROUTE_IDS
} from '@/components/routes/training-plan'
import {getPathById} from '@/components/routes/utils'
import useEventResults from '@/hooks/event/use-event-results'
import EventResultsInjector from '@/injectors/event/ResultsInjector'

import messages from './messages'
import createRoutes from './modal-routes'

import {startOfDay, startOfWeek, addDays, format, isSameDay} from 'date-fns'

import cx from 'classnames'

import {de} from 'date-fns/locale'

// import Calendar from '@/components/molecules/Calendar'

import DateCyclePicker from '@/components/molecules/DateCyclePicker'

import EventCard from './EventCard'
// import {EVENT_COLORS} from '@/modules/event/results/types'

// <FontAwesomeIcon icon="fa-solid fa-image" />
// <FontAwesomeIcon icon="fa-solid fa-circle-play" />
// <FontAwesomeIcon icon="fa-solid fa-file-lines" />
// <FontAwesomeIcon icon="fa-solid fa-video" />

// <FontAwesomeIcon icon="fa-regular fa-clock" />

// <FontAwesomeIcon icon="fa-solid fa-up-from-line" />

// import {MODAL_COLORS} from '@/components/atoms/modal'

import mockedEventData from './mockEventData'

const BASE_PATH = getPathById([
  ROUTE_IDS.HOME,
  TRAINING_PLAN_BRANCH_ID,
  TRAINING_PLAN_ROUTE_IDS.ROOT
])

const ROUTES = createRoutes(BASE_PATH)

const TrainingPlan = () => {
  const now = startOfDay(new Date())
  const startOfThisWeek = startOfWeek(now, {weekStartsOn: 1})

  const [state, events] = useEventResults({
    id: TRAINING_PLAN_MODULE_ID,
    shouldLoadOnMount: true
  })
  const [selectedDate, setSelectedDate] = useState(startOfThisWeek)

  const handleChangeDate = useCallback(date => {
    setSelectedDate(date)
  }, [])

  const handleStartDateChange = useCallback(newDate => {
    setSelectedDate(startOfWeek(newDate, {weekStartsOn: 1}))
  }, [])

  const startDate = startOfWeek(selectedDate, {weekStartsOn: 1})
  const endDate = addDays(startDate, 6)
  const updatedDays = Array(7)
    .fill()
    .map((_, i) => addDays(startDate, i))

  const hasEventsForDay = (day, events) => {
    return events.some(event => isSameDay(new Date(event.date), day))
  }

  return (
    <ModalRouter base={BASE_PATH} routes={ROUTES}>
      <EventResultsInjector />
      <div className="h-full flex flex-col overflow-hidden">
        <div>
          <PageHeader
            className="relative z-20"
            title={messages.pageTitle}
            dateCyclePicker={
              <DateCyclePicker
                value={startDate}
                onChange={handleStartDateChange}
              />
            }
          />
        </div>

        <div className="flex flex-col h-full ">
          <div className="flex sticky top-0 z-10 bg-secondary mx-4">
            {updatedDays.map(day => (
              <div
                key={day.getTime()}
                className={cx(
                  'flex-1 p-3 border-solid border-x border-gray-300 shadow',
                  {'bg-white': day.getDate() === now.getDate()}
                )}
              >
                <div className="flex">
                  <div className="font-bold text-[16px] ml-2 ">
                    {format(day, 'EEEE', {locale: de})
                      .substring(0, 2)
                      .toUpperCase()}
                  </div>
                  <div className="text-[16px] ml-2 relative">
                    {format(day, 'dd.MM.')}
                  </div>
                </div>
                {isSameDay(day, now) && (
                  <div className="relative">
                    <div className="absolute -inset-x-1 bottom-5 h-1 rounded-sm bg-blue-900" />
                  </div>
                )}
              </div>
            ))}
          </div>
          <div className="flex flex-row h-full mx-4 overflow-y-auto ">
            {updatedDays.map((day, index) => (
              <div
                key={day.getTime()}
                className={`bg-secondary border border-gray-300 p-0.5 flex-1 h-full overflow-auto relative ${
                  day.getDate() === now.getDate() ? 'bg-white' : ''
                }`}
                style={{
                  overflowY: 'scroll',
                  WebkitScrollbar: {display: 'none'},
                  msOverflowStyle: 'none',
                  scrollbarWidth: 'none'
                }}
              >
                {/* evencard with mock data */}
                {mockedEventData.map(event => (
                  <EventCard key={event.id} event={event} day={day} />
                ))}
              </div>
            ))}
          </div>
        </div>
      </div>
    </ModalRouter>
  )
}

export default React.memo(TrainingPlan)
