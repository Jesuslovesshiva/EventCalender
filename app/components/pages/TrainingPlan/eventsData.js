import React from 'react'
import EventCard from './EventCard'

export const eventsData = [
  {
    day: 1, // Monday
    event: {
      time: '10:00',
      title: 'Training',
      attendees: 15,
      activities: [
        {
          title: 'Krafttraining Beine',
          duration: 30,
          maxParticipants: 20,
          description: 'Basis Bewegung Verbesserung der Bein-Kraft, QualitÃ¤t'
        },
        {
          title: 'Hands-on',
          duration: 60,
          maxParticipants: 20,
          description: 'hehe'
        }
      ],
      attachments: [{type: 'pdf'}, {type: 'jpg'}, {type: 'mp4'}]
    }
  }
  // ... more events
]

const renderEventCard = dayIndex => {
  // Get the events for the current day
  const dayEvents = events.filter(
    event =>
      format(new Date(event.startDate), 'yyyy-MM-dd') ===
      format(days[dayIndex - 1], 'yyyy-MM-dd')
  )

  // Render the events for the current day
  return dayEvents.map(event => (
    <EventCard
      key={event.id}
      time={format(new Date(event.startDate), 'HH:mm')}
      title={event.title}
      attendees={event.participantsCount + event.guestParticipantsCount}
      activities={event.exercises}
      attachments={event.files}
    />
  ))
}
