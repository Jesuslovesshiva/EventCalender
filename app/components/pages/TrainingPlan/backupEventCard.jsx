import React from 'react'
import {format} from 'date-fns'
import {isSameDay} from 'date-fns/esm'
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
import icons from './icons'
// import {icon} from '@fortawesome/fontawesome-svg-core'
// import {fontSize} from 'tailwindcss/defaultTheme'

// import {MODAL_COLORS} from '@/components/atoms/Modal'
// import {EVENT_TYPES} from '@/modules/event/results/types'

const getBackgroundColorForEventType = eventType => {
  switch (eventType) {
    case 'covid':
      return 'bg-gradient-to-b from-amber-100 to-white'
    case 'heimspiel':
      return 'bg-gradient-to-b from-rose-200 to-white'
    default:
      return 'bg-gray-200'
  }
}

function LocationBadge({location, locationIcon}) {
  return (
    <div className="bg-gray-300 border border-gray-400 rounded-md p-1 px-2 font-thin text-[15px] mr-3 flex ">
      {location} <FontAwesomeIcon icon={locationIcon} className="ml-1" />
    </div>
  )
}

function ParticipantsBadge({participantsCount, participantsIcon}) {
  return (
    <div
      className={`rounded-xl bg-white p-1 px-2 shadow flex mx-2 ${
        participantsCount ? 'space-x-1' : ''
      }`}
      style={{maxWidth: '80px'}}
    >
      <FontAwesomeIcon icon={participantsIcon} className="text-[14px] " />
      <p className="font-bold text-[14px] ">{participantsCount}</p>
    </div>
  )
}

function DownloadButton({upload}) {
  return (
    <div
      className={`rounded-md p-1 px-2 shadow flex mx-2 border-gray-400 border`}
    >
      <FontAwesomeIcon icon={upload} className="text-[14px] " />
    </div>
  )
}

function EventCard({event, day}) {
  if (!isSameDay(event.start, day)) {
    return null
  }

  const {
    location,
    participantsCount,
    locationIcon,
    participantsIcon,
    type,
    exercises,
    upload
  } = event

  const backgroundColor = getBackgroundColorForEventType(type)

  return (
    <div
      className={`event-card ${backgroundColor} border-white border-2 rounded-md shadow mb-3 m-1 overflow-hidden`}
    >
      {exercises.map((exercise, index) => (
        <React.Fragment key={`${exercise.name}-${index}`}>
          {index === 0 && (
            <>
              <div className="flex items-center flex-wrap mb-1 border-b-2 border-gray-300 ">
                <div className="flex justify-between items-center space-x-2 m-3 pb-1 w-full">
                  <div className="flex items-center space-x-2">
                    {type == 'heimspiel' ? (
                      <div className="bg-red-100 border border-red-200 rounded-md p-1 font-thin text-[15px]">
                        {format(event.start, 'HH:mm')}
                      </div>
                    ) : null}
                    {type == 'training' ? (
                      <div className="bg-gray-300 border border-gray-400 rounded-md p-1 font-thin text-[15px]">
                        {format(event.start, 'HH:mm')}
                      </div>
                    ) : null}
                    {type == 'covid' ? (
                      <div className="bg-yellow-100 border border-yellow-200 rounded-md p-1 font-thin text-[15px]">
                        {format(event.start, 'HH:mm')}
                      </div>
                    ) : null}
                    <div className="font-bold text-[21px] ">{event.title}</div>
                  </div>

                  {type !== 'heimspiel' && type !== 'covid' ? (
                    <LocationBadge
                      location={location}
                      locationIcon={locationIcon}
                    />
                  ) : null}
                </div>
                {exercise.descriptionPosition === 'afterTitle' && (
                  <div className="text-lg p-3 -mt-3 w-full">
                    {exercise.description}
                  </div>
                )}
              </div>
            </>
          )}

          {index === 0 && (
            <div className="flex justify-between items-center m-1">
              <ParticipantsBadge
                participantsCount={participantsCount}
                participantsIcon={participantsIcon}
              />
              <div className="flex justify-end items-between">
                {type !== 'covid' ? (
                  <div className="rounded-xl bg-white p-1 shadow flex px-2 space-x-1 ">
                    {index === 0 &&
                      exercises
                        .reduce((acc, exercise) => {
                          ;(exercise.files || []).forEach(fileIcon => {
                            if (
                              !acc.find(
                                icon => icon.iconName === fileIcon.iconName
                              )
                            ) {
                              acc.push(fileIcon)
                            }
                          })
                          return acc
                        }, [])
                        .map((icon, idx) => (
                          <FontAwesomeIcon
                            key={idx}
                            icon={icon}
                            className="text-[14px]"
                          />
                        ))}
                  </div>
                ) : null}

                <DownloadButton upload={upload} />
              </div>
            </div>
          )}

          {type !== 'heimspiel' && type !== 'covid' ? (
            <div className="border-t-2 pt-2 pb-4 border-gray-300">
              <div className="h-4/5 ">
                <div className="flex items-center mb-2">
                  <div className="p-1.5 text-[16px] font-bold ml-2">
                    {index + 1}.
                  </div>
                  <div className=" text-[16px] font-bold">{exercise.name}</div>
                </div>

                <div className="flex justify-between items-center m-1 mx-2  ">
                  <ParticipantsBadge
                    participantsCount={exercise.participantsCount || 0}
                    participantsIcon={participantsIcon}
                  />
                  {exercise.files && exercise.files.length > 0 && (
                    <div
                      className="rounded-xl bg-white p-1 shadow flex px-2 space-x-1"
                      style={{maxWidth: '80px'}}
                    >
                      {exercise.files.map((icon, idx) => (
                        <FontAwesomeIcon
                          key={idx}
                          icon={icon}
                          className="text-[14px]"
                        />
                      ))}
                    </div>
                  )}
                </div>

                <div className="p-1 flex px-6 mt-2 space-x-1">
                  <FontAwesomeIcon
                    icon={icons.target}
                    className="text-[16px]"
                  />
                </div>
                {exercise.descriptionPosition !== 'afterTitle' && (
                  <div className=" -mt-5 ml-12 mr-6 text-lg ">
                    {exercise.description}
                  </div>
                )}
                {index !== exercises.length - 1 && (
                  <div className="event-card bg-gray-200  "></div>
                )}
              </div>
            </div>
          ) : null}
        </React.Fragment>
      ))}
    </div>
  )
}

export default EventCard
