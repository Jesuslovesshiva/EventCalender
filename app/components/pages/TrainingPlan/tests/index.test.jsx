import {act} from '@testing-library/react'

import ToastContainer from '@/components/atoms/ToastContainer'
import {TRAINING_PLAN_MODULE_ID} from '@/components/pages/TrainingPlan/constants'
import EntitiesInjector from '@/injectors/entities/EntitiesInjector'
import EventResultsInjector from '@/injectors/event/ResultsInjector'
import {actions} from '@/modules/event/results'
import EVENT_TYPES from '@/modules/event/results/types'
import configureTestStore from '@/utils/test-utils/configure-test-store'
import render from '@/utils/test-utils/custom-render'
import renderInjectors from '@/utils/test-utils/render-injectors'
import setEntities from '@/utils/test-utils/set-entities'

import TrainingPlan from '../index'

jest.mock('@tanstack/react-location', () => ({
  ...jest.requireActual('@tanstack/react-location'),
  useNavigate: jest.fn(() => {})
}))
jest.mock('date-fns', () => ({
  ...jest.requireActual('date-fns'),
  startOfWeek: jest.fn(() => new Date('2022/01/31 08:00:00'))
}))
jest.mock('@/components/molecules/ModalRouter')

const {clearDispatch, options, store} = configureTestStore({
  withActionSubscription: true
})
const renderComponent = (renderFn = render) =>
  renderFn(
    <>
      <ToastContainer />
      <TrainingPlan />
    </>,
    {
      ...options,
      wrapperProps: {
        hasDndProvider: true,
        hasRouter: true,
        ...options.wrapperProps
      }
    }
  )

const event = {
  id: 'event-id-1',
  startedAt: new Date('2022-01-31 08:00:00').toISOString(),
  title: 'Event Title',
  type: EVENT_TYPES.OTHER
}

const dispatchResult = ({result = [event.id], total = 1}) => {
  act(() => {
    store.dispatch(
      actions.load.fulfilled({
        id: TRAINING_PLAN_MODULE_ID,
        result,
        total
      })
    )
  })
}

describe('TrainingPlan', () => {
  beforeAll(() => {
    jest.useFakeTimers().setSystemTime(new Date('2022/01/31').getTime())
    renderInjectors(
      <>
        <EventResultsInjector />
        <EntitiesInjector />
      </>,
      options
    )
    setEntities(store, {
      events: {
        [event.id]: event
      }
    })
  })

  beforeEach(() => {
    clearDispatch()
  })

  afterEach(() => {
    store.dispatch(actions.reset({id: TRAINING_PLAN_MODULE_ID}))
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  it('should match the snapshot', () => {
    const {container} = renderComponent()
    dispatchResult({})
    expect(container).toMatchSnapshot()
  })
})
