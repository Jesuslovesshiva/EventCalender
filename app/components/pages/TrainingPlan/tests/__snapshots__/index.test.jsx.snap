// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingPlan should match the snapshot 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="h-full flex flex-col"
  >
    <div
      class="grid grid-cols-3 items-center px-6 py-4 bg-gradient-to-b bg-gradient-silver shadow relative z-20"
    >
      <div
        class="flex items-center"
      >
        <h1
          class="shrink-0 grow-0 mr-3 font-page-heading text-primary"
        >
          Trainingsplan
        </h1>
      </div>
      <div
        class="justify-self-center"
      />
      <div
        class="justify-self-end"
      />
    </div>
    <div
      class="grow overflow-auto"
    >
      <h2
        class="font-heading-large"
      >
        Have fun :)!
      </h2>
    </div>
  </div>
</div>
`;
